src/rank_llm/retrieve/retriever.py

```diff
@@ -84,15 +91,38 @@ def from_saved_results(file_name: str):
             )
         retriever = Retriever(RetrievalMode.SAVED_FILE, dataset=retrieved_results)
         return retriever.retrieve()
+    
+    @staticmethod
+    def from_custom_index(index_path: str, topics_path: str, index_type: str):

```

Comment:
pyserini's init from custom index gets encoder: str = None, onnx: bool = False params that this method doesn't have it. If the default values work, please add them with their default values to from_custom_index` function as well, if not then these parameters should be added to `from_custom_index` and raise value errors if they are not received from the user.